uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -212,40 +293,51 @@ define([
      * @return {Promise} of array with day types
      */
     function filterLeaveRequestDayTypes (date, dayType) {
-      var listToReturn;
-
       if (!date) {
         return $q.reject([]);
       }
 
-      // Make a copy of the list
-      listToReturn = vm.requestDayTypes.slice(0);
-      date = vm._convertDateToServerFormat(date);
+      date = convertDateToServerFormat(date);
 
       return PublicHoliday.isPublicHoliday(date)
         .then(function (result) {
           if (result) {
-            return listToReturn.filter(function (publicHoliday) {
+            return vm.requestDayTypes.filter(function (publicHoliday) {
               return publicHoliday.name === 'public_holiday';
             });
           }
 
-          return getDayTypesFromDate(date, listToReturn)
+          return getDayTypesFromDate(date, vm.requestDayTypes)
             .then(function (inCalendarList) {
               return inCalendarList.length
                 ? inCalendarList
-                : listToReturn.filter(function (dayType) {
+                : vm.requestDayTypes.filter(function (dayType) {
                   return _.includes(['all_day', 'half_day_am', 'half_day_pm'], dayType.name);
                 });
             });
         })
-        .then(function (listToReturn) {
-          setDayType(dayType, listToReturn);
+        .then(function (dayTypes) {
+          setDayTypes(dayType, dayTypes);
 
-          return listToReturn;
+          return dayTypes;
         });
     }
 
+    /**
+     * Gets absence period for the given date
+     *
+     * @param  {String} dateType from|to
+     * @return {Object} Absence period instance or empty object if not found
```

Comment:
this function should throw if no period is found, i don't see any reason why it shouldn't. In fact you are doing exactly that in one of the callers, `dateChangeHandler`, so do it directly in here. It makes sense as a whole, you would not expect this function to return an empty object if it can't find anything, you would expect it to throw an error

Reply:
@AkA84 While I totally agree that it might be improved, I suggest to postpone this. The reason is that to accomplish this change we will need to address the following:
1) I don't think it should throw an exception if absence period is not found, it could reject a promise, but `getAbsencePeriod` is not a promise. So the chain rejection should happen after.
2) absence period still needs to set as {} - if we do not want it to be {}, we should refactor the rest of the code.